osquery/tables/system/linux/process_open_pipes.cpp

```diff
@@ -0,0 +1,193 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <iostream>
+#include <osquery/core.h>
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <regex>
+
+namespace osquery {
+namespace tables {
+
+struct pipe_info {
+  pid_t pid;
```

Comment:
More of a style nit, but I suggest using `pid_` and similar for C++ members. Alternatively, we could probably avoid a custom struct entirely and do something like:

```cpp
using pipe_info = std::tuple<pid_t, int, std::string, ino_t, std::string>;
```